### ISSUE TITLE
create perpendicular line

### ISSUE THREAD

**Issue by wykhuh** [LibreCAD/LibreCAD#731]
Hi, I just started using LibreCAD. I would like to create a perpendicular line from a selected point to a selected line. In the screenshot, I want to draw a line from Point 1 that is perpendicular to line  B

<img width="465" alt="screen shot 2016-05-29 at 12 49 09 pm" src="https://cloud.githubusercontent.com/assets/6968611/15635741/d12f1a28-259b-11e6-9944-4f7cce4ca5b6.png">

In AutoCAD,  there is a snap to [perpendicular option](http://www.mycadsite.com/tutorials/level_4/autocad-osnaps-4-2.htm).

 Is there a similar function in LibreCAD?  In order to draw the perpendicular line, I had to
1) use `perp` to draw a line perpendicular to line B, place the midpoint of the perpendicular line at point 
2) git rid of the  left half of the perpendicular line at point one by using `trim`
3) extend the right half of the  perpendicular line to line B by using `trim`

**Comment by r-a-v-a-s** [comment]
It's a useful feature that we don't have...
http://forum.librecad.org/New-contributors-amp-The-LibreCAD-team-td5713777.html

Your trim solution might be the optimal one...
I've been setting the orthogonal line length to 10000
and then I use the snap intersection mode to create the desired line,
and delete the one created with the orthogonal tool.

**Comment by wykhuh** [comment]
I would like to help with the documentation. Searching through Github issues, `rs_commands.cpp`,  and the forum on how to do stuff is not the best use of time. Any suggestions on how I can get started?  I never worked in C, so I won't be able to contribute to the codebase quite just yet.

**Comment by lordofbikes** [comment]
Yes, it's very similar in LibreCAD, use the orthogonal line tool for this.

1. With the lines A and B drawn use the Orthogonal line tool from the Tools->Line menu or use command `perp` or `ortho`, the cursor changes to a gray cross.
2. Somewhere on the top the tool specific bar appear where you can set the length for the orthogonal line. Change it to your needs.
3. Click on line B to set it as reference for the orthogonal line. The color of line B changes to visualize this.
4. Use a valid snap mode an click on point 1 to place the line. The orthogonal line is placed with it's center in point 1, orthogonal to line B.
  * You can place more orthogonal lines relative to line B until you click the right button or press ESC.
  * Then you can select a new reference for more orthogonal lines
  * or click right or press ESC again to cancel the orthogonal line tool
5. Now use Trim or Trim two to trim the drawing for your needs.

**Comment by r-a-v-a-s** [comment]
If you want to help complete http://wiki.librecad.org/index.php/Commands
that would be awesome...

When I rewrote it I made the table formatting consistent.
I don't think you will have a problem seeing the pattern.

The main page has a 'wiki help' section with links.
http://wiki.librecad.org/

Here is the majority of new additions:
http://forum.librecad.org/updating-rs-commands-td5711646.html#a5711654

Feel free to post in the dev section regarding working on documentation.
http://forum.librecad.org/

**Comment by r-a-v-a-s** [comment]
I'm going to assume you decided against doing the work.

**Comment by fa201** [comment]
@lordofbikes : feature is done and reference documentation is updated.
